### PR TITLE
fix: optimize chat list loading [ch17654]

### PR DESCRIPTION
**Pull request by dchest**
#### Relevant info and issue/PR links 

https://app.clubhouse.io/peerio/story/17654/fix-one-by-one-loading-and-redrawing-of-chat-list
 
#### Testing instructions  

1. Make sure chats load.
2. On Android, check if initial chat list loading is not loading rooms one-by-one as it was before.

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

**Comment by dchest** [comment]
@seavan I have a suspicion that mobile doesn't wait for `chatStore.loaded` (or `!chatStore.loading`) during the loading screen. I've checked but couldn't find the place. Could you please check it?

File: src/models/chats/chat-store.ts [hunk 1]
```diff
@@ -145,6 +146,7 @@ export class ChatStore {
      */
     @computed
     get channels() {
+        if (!this.loaded) return [];
```
Comment (dchest) [inline src/models/chats/chat-store.ts]: Before this during initial load it was recalculating `channels` every time the new chat head loads, triggering re-render. Now it waits until `this.loaded` before it returns all loaded chats. Note that we have a timeout to waiting before `this.loaded` is true in `loadAllChats`, so it won't wait for all chats to load if it takes more than  5 seconds.

Comment (anri-asaturov) [reply src/models/chats/chat-store.ts]: yeah this is fine

Comment (seavan) [reply src/models/chats/chat-store.ts]: @dchest yes it does wait during the loading screen:

it waits for chatState.init which waits for chatStore.loaded here:

https://github.com/PeerioTechnologies/peerio-mobile/blob/chore-login-refactor/app/components/messaging/chat-state.js#L53

**Pull request merged**